internal/controllers/bundledeployment/bundledeployment.go

```diff
@@ -101,6 +101,12 @@ func WithActionClientGetter(acg helmclient.ActionClientGetter) Option {
 	}
 }
 
+func WithPreflights(preflights ...Preflight) Option {
```

Comment:
Nit: Is this required and present by default for now? If so, it would be nice to add a validation for this in `validateConfig` func, just to avoid programmer's error since we are calling `preflight.Install` directly.

Reply:
It is not required. If there are no preflight checks defined neither `preflight.Install()` nor `preflight.Upgrade()` should be called.